stylix/testbed.nix

```diff
@@ -0,0 +1,95 @@
+{ pkgs, inputs, lib, ... }:
+
+let
+  commonModule = { config, ... }: {
+    users.users.testbed = {
+      description = "Testbed";
+      hashedPassword = "";
+      isNormalUser = true;
+    };
+
+    # The state version can safely track the latest release because the disk
+    # image is ephermal.
+    system.stateVersion = config.system.nixos.release;
+    home-manager.users.testbed.home.stateVersion = config.system.nixos.release;
+
+    virtualisation.vmVariant.virtualisation = {
+      # This is a maximum limit; the VM should still work if the host has fewer cores.
+      cores = 4;
+      memorySize = 2048;
```

Comment:
Since each testbed has its own NixOS module, it's already possible to override this value on an individual basis:

```nix
{ lib, ... }:

{
  # Install the target ... then:

  virtualisation.vmVariant.virtualisation.memorySize = lib.mkForce 4096;
}
```

The current value is double the default, and should be just enough for any of the testbeds added in this PR.

Reply:
> Since each testbed has its own NixOS module, it's already possible to override this value on an individual basis:
> 
> ```nix
> { lib, ... }:
> 
> {
>   # Install the target ... then:
> 
>   virtualisation.vmVariant.virtualisation.memorySize = lib.mkForce 4096;
> }
> ```
> 
> The current value is double the default, and should be just enough for any of the testbeds added in this PR.

In that case, the API would be better if the back-end (`stylix/testbed.nix`) uses `lib.mkDefault` and the front-end (individual testbeds) simply uses:

```suggestion
      memorySize = lib.mkDefault 2048;
```

Reply:
Absolutely.